src/microbe_stage/organelle_components/PilusComponent.cs

```diff
@@ -45,7 +47,8 @@ public override void OnShapeParentChanged(Microbe newShapeParent, Vector3 offset
 
             // New ownerId
             var ownerId = currentShapesParent.CreateNewOwnerId(newShapeParent, transform, addedChildShapes[0]);
-            newShapeParent.AddPilus(ownerId);
+            var isInjectisome = organelle.Upgrades?.UnlockedFeatures.Contains(PILUS_INJECTISOME_UPGRADE_NAME);
+            newShapeParent.AddPilus(ownerId, isInjectisome ?? false);
```

Comment:
```suggestion
            newShapeParent.AddPilus(ownerId, isInjectisome == true);
```

This is my preferred way for checking if a nullable bool is set.